lib/apt_update_nonblocking_hook.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/python3
+
+import logging
+from datetime import datetime, timedelta, timezone
+
+from uaclient import apt
+from uaclient.apt_news import fetch_and_process_apt_news
+from uaclient.config import UAConfig
+from uaclient.daemon import setup_logging
+
+
+def main(cfg: UAConfig):
+    if cfg.apt_news:
+        last_update = apt.get_apt_cache_datetime()
+        one_day_ago = datetime.now(timezone.utc) - timedelta(days=1)
+        if last_update is None or last_update < one_day_ago:
+            fetch_and_process_apt_news(cfg)
```

Comment:
@basak this is where the additional checks happen to avoid unnecessary apt_news requests.

`if cfg.apt_news` is what respects the `pro config set apt_news=false` setting

the date logic compares "now" to the timestamp of the apt-update stamp file. If the apt-update stamp is within the past 24 hours, we don't do anything here.

LMK what you think